File: BmobPushDemoSend/BmobPushDemoSend/ViewController.m
//
//  ViewController.m
//  BmobPushDemoSend
//
//  Created by limao on 15/5/6.
//  Copyright (c) 2015年 Bmob. All rights reserved.
//

#import "ViewController.h"
#import <BmobSDK/Bmob.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)sendMsgUseBoardcastBtn:(UIButton *)sender {
    BmobPush *push = [BmobPush push];
    //设置推送消息
    [push setMessage:@"所有人的推送的消息"];
    //发送推送
    [push sendPushInBackgroundWithBlock:^(BOOL isSuccessful, NSError *error){
        if (isSuccessful) {
            NSLog(@"boardcast successful");
        } else if (error){
        NSLog(@"error %@",[error description]);
        }
    }];
}


@end

File: SwiftDemo/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by Bmob on 16/5/3.
//  Copyright © 2016年 bmob. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

//        queryUsers()

//        save()

//        update()

        deleteGameScore()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func queryUsers()  {
        let query:BmobQuery = BmobUser.query()
        query.order(byDescending: "createdAt")
        query.findObjectsInBackground { (array, error) in
            for i in 0..<array?.count{
                let obj :BmobUser = (array[i] as? BmobUser)!
                print("object id \(obj.objectId),username \(obj.username)")

            }
        }
    }

    func save(){
        let gamescore:BmobObject = BmobObject(className: "GameScore")
        gamescore.setObject("Jhon Smith", forKey: "playerName")
        gamescore.setObject(90, forKey: "score")
        gamescore.saveInBackground { (isSuccessful, error) in
            if error != nil{
                print("error is \(error?.localizedDescription)")
            }else{
                print("success")
            }
        }
    }

    func update() {
        let  gamescore:BmobObject = BmobObject(outDatatWithClassName: "GameScore", objectId: "f3a82207ed")
        gamescore.setObject(91, forKey: "score")
        gamescore.updateInBackground { (isSuccessful, error) in
            if error != nil{
                print("error is \(error?.localizedDescription)")
            }else{
                print("success")
            }
        }
    }

    func deleteGameScore()  {
        let  gamescore:BmobObject = BmobObject(outDatatWithClassName: "GameScore", objectId: "4faf28f4dd")
        gamescore.deleteInBackground { (isSuccessful, error) in
            if error != nil{
                print("error is \(error?.localizedDescription)")
            }else{
                print("success")
            }
        }
    }
}

